CAIPs/caip-50.md

```diff
@@ -0,0 +1,207 @@
+---
+caip: 50
+title: Multi-Chain Account ID Specification
+author: Joel Torstensson (@oed), Pedro Gomes (@pedrouid)
+discussions-to: https://github.com/ChainAgnostic/CAIPs/pull/50
+status: Draft
+type: Standard
+created: 2020-06-10
+updated: 2020-06-11
+requires: 2
+---
+
+## Simple Summary
+
+CAIP-50 defines a way to identify blockchain account addresses uniquely across multiple blockchains
+
+## Abstract
+
+This proposal aims to facilitate using unique address for accounts on multiple blockchain systems using a multi-codec format that encodes variable integer for chain identifiers complaint with CAIP-2 blockchain id specification. This is useful for both decentralized applications and wallets to communicate user accounts for multiple chains using unique identifiers which are machine veriable and can be decoded to identify the original address and chainId encoded. This proposal aims to standardize these identifiers for accounts to allow inteoperability for multi-chain applications.
+
+## Motivation
+
+The motivation for this proposal came from different feedback received from the adoption of CAIP-10 and tackles these to provide a significant improvement in more efficient communication of these identifiers using smaller byte footprint while preserving uniqueness and interoperability.
+
+## Specification
+
+In this specifiation we define the Multi-Chain Account Id or MACI for short which is a compactly encoded account identifier that is contextual to a blockchain using a CAIP-2 chainId.
+
+### Syntax
+
+```js
+mcai ::= <multibase_prefix><mcai_code><chain_namespace><id_size><chain_id><address_size><address><parity_byte>
+```
+
+### Semantics
+
+- `multibase_prefix` - the prefix which defines which multibase is used to encode the bytes, `z` for `base58btc`
+- `mcai_code` - a number registered on the [multicodec table](https://github.com/multiformats/multicodec/blob/master/table.csv), makes the multi-chain account id upgradable, encoded as varint
+- `chain_namespace` - see table below, encoded as varint
+- `id_size` - the length in bytes of the `chain_id`, encoded as varint
+- `chain_id` - the chain id, encoding is defined by the chain namespace
+- `address_size` - the length of the address, encoded as varint
+- `address` - the address itself, encoding is defined by the chain namespace
+- `parity_byte` - a checksum byte, see section below
+
+### MCAI multicodec
+
+Should be a number registered on the [multicodec table](https://github.com/multiformats/multicodec/blob/master/table.csv). In the examples below we use `0xCA` but this is subject to change.
+
+### Chain namespaces
+
+Each blockchain namespace needs to be properly defined with a registry table:
+
+#### Registry Table
+
+| Namespace | code |
+| --------- | ---- |
+| bip122    | 0x00 |
+| eip155    | 0x01 |
+| cosmos    | 0x02 |
+| polkadot  | 0x03 |
+| filecoin  | 0x04 |
+| lip9      | 0x05 |
+| eosio     | 0x06 |
+| tezos     | 0x07 |
```

Comment:
Perhaps the Registry would go into its own CAIP

Reply:
I agree with you Pedro. It would be make a lot of sense.

Reply:
not sure it helps, but if it would increase the sense of legitimacy in some way, we could host it on github.com/multiformats/ with ownership by the relevant people here to manage it